_posts/2023-02-07-Announcing-Data-Prepper-2.1.0.md

```diff
@@ -0,0 +1,253 @@
+with `aggregate_when` option. We can also generate a histogram of trace latencies using another aggregate processor in 
+a parallel sub-pipeline.
+
+The pipeline configuration for the above example would look like
+
+```
+trace-error-metrics-pipeline:
+  source:
+    pipeline:
+      name: "span-pipeline"
+  processor:
+    - aggregate:
+        identification_keys: ["serviceName", "traceId"]
+        action:
+          count:
+        group_duration: "20s"
+        aggregate_when: "/status_code == 2"
+  sink:
+     - opensearch
+         hosts: ["https://opensearch:9200"]
+        insecure: true
+        username: "admin"
+        password: "admin"
+        index: trace-error-metrics
+        
+ trace-high-latency-metrics-pipeline:
+  source:
+    pipeline:
+      name: "span-pipeline"
+  processor:
+    - aggregate:
+        identification_keys: ["serviceName", "traceId"]
+        action:
+          histogram:
+            key: "durationInNaos"
+            record_minmax: true
+            units: "nanoseconds"
+            buckets: [1000000000, 1500000000, 2000000000]
+        group_duration: "20s"
+        aggregate_when: "/durationInNanos > 1000000000"
+  sink:
+      - opensearch
+         hosts: ["https://opensearch:9200"]
+         insecure: true
+         username: "admin"
+         password: "admin"
+         index: trace-high-latency-metrics
+```
+
+## Anomaly detection
+
+Data Prepper 2.1 introduces anomaly detection as an independent processor that can be placed anywhere in the pipeline to 
+detect anomalies using AI/ML techniques. It supports only random cut forest algorithm to detect anomalies for now. 
+Support for other AI/ML algorithms may be added in future. The processor can be configured to use any numerical (integer 
+or floating point data type) field in the input events to detect anomalies.
+
+Anomaly detection in Data Prepper allows for more scalable and efficient way to detect anomalies in the logs and traces 
+because the ML algorithm is applied on the server side.
+
+For example, the anomaly detector processor placed after aggregate processor that generates histogram of trace latencies 
+(`durationInNanos`) could detect abnormally high latencies. The following anomaly detector processor configuration 
+can be used for it.
+
+```
+trace-metric-anomaly-detector-pipeline:
+  source:
+    pipeline:
+      name: "trace-high-latency-metrics-pipeline"
+  processor:
+    - anomaly_detector:
+        keys: ["max"]
+        mode:
+          random_cut_forest:
+  sink:
+    - opensearch
+         hosts: ["https://opensearch:9200"]
+         insecure: true
+         username: "admin"
+         password: "admin"
+         index: high-latency-traces-with-anomalies
+```
+
+## Sampling and limiting
+
+Data Prepper 2.1 supports sampling and rate-limiting to limit the number of events that are sent to a Sink.
+
+One possible use case for both these features can be used to reduce the load on the OpenSearch when storing normal 
```

Comment:
```suggestion
One possible use case for both these features is to reduce the load on the OpenSearch when storing normal 
```

Reply:
Could we say "ordinary" instead of "normal"?